### PR TITLE
New component recaptcha

### PR DESCRIPTION
**Pull request by Ian-c-cameron**
Replacement for captcha component to use Google Recaptcha v2

### CODE REVIEW

**Review by harrymaynard-maximus — CHANGES_REQUESTED** [review]
One comment on the `siteKey`.

Question: does the common library still work with other projects that haven't imported the "ng-recaptcha" NPM package?

Also, I think Jam and everyone else should see a demo of this in action before merging it into the library.

File: projects/common/recaptcha/src/recaptcha.component.html [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<re-captcha (resolved)="resolved($event)" siteKey="6LfHcJcbAAAAAJA_kkeR4AXt92hSUpCxb-mKeWkT">
```
Comment (harrymaynard-maximus) [inline projects/common/recaptcha/src/recaptcha.component.html:1]: We definitely don't want to store the `siteKey` in the source code. It should be passed to the component from an environment variable from "spa-env-server".

Comment (Ian-c-cameron) [reply projects/common/recaptcha/src/recaptcha.component.html:1]: The siteKey is a public key.  The public key is in the UI component, and the private key is an env variable to the server side recaptcha-service.
I can't fully demo the final comunication with the recaptcha service until it is in the openshift environment, which requires it to be integrated into an appication(BCP).  To do this it must be in the main branch of the common library.

Comment (harrymaynard-maximus) [reply projects/common/recaptcha/src/recaptcha.component.html:1]: 1. So does that mean it's ok to share the same `siteKey` for all projects? I thought Google limits the number of captcha requests/verifications each key can have before they start charging money for it's use. If that's the case, then we probably want the ability to have different public keys too.

2. Ok, but does MSP for example still run with the library linked to it locally?

Comment (Ian-c-cameron) [reply projects/common/recaptcha/src/recaptcha.component.html:1]: 1) we are likely to need to pay for a key set, but we will need to discuss that with Jam and work out how we want it to work for licensing.
2) MSP still works with no changes when linked to the library locally.
The recaptcha component displays and workes to the point of attempted communictation with the service on my branch of BCP when linked locally, but until it is on openshift I can't test the link between component and service.

Comment (harrymaynard-maximus) [reply projects/common/recaptcha/src/recaptcha.component.html:1]: Ok thanks, sounds good.

**Review by harrymaynard-maximus — APPROVED** [review]

**Review by bryce-mcmath — APPROVED** [review]
Nice work

File: projects/common/recaptcha/src/recaptcha-data.service.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+// import { Http, Response } from '@angular/http';
```
Comment (harrymaynard-maximus) [inline projects/common/recaptcha/src/recaptcha-data.service.ts:3]: Can you remove unused imports please.

**Review by harrymaynard-maximus — CHANGES_REQUESTED** [review]

File: projects/common/recaptcha/src/recaptcha-data.service.ts (hunk 2 above)
Comment (Ian-c-cameron) [reply projects/common/recaptcha/src/recaptcha-data.service.ts]: I only see the one commented import that is unused.
I have removed it.

Thanks,
Ian

**Review by harrymaynard-maximus — APPROVED** [review]
Looks good.

**Pull request merged**